### PR TITLE
feat: generate and upload crowdin glossary

### LINKED ISSUE #82: Build a glossary
Now that we're [including Electron's markdown API docs](https://github.com/electron/electron-i18n/issues/67) in the source content, there are pieces of those docs that we don't want translated. Examples include api names like `BrowserWindow`, method names like `win.isFocused`, data types like `String` and `Array`, etc. 

Here's an example of what we **don't want**:

screen shot 2017-09-11 at 1 38 34 pm

We can use Crowdin's glossaries to solve this problem. From Crowdin support:

> Glossaries are created for the clarification of the specific terms used in the project. Terms are displayed as the underlined words in the Editor.
>
> They get highlighted while hovering on them so that the project participants can see the description.
>
> In the description of the term you can specify that *String should not be translated because it's a data type*.
>
> You may upload/download glossaries via API if you have any existing ones. Here more information:
> 
> https://support.crowdin.com/api/upload-glossary/
> https://support.crowdin.com/api/download-glossary/

### PR DESCRIPTION
**Pull request by zeke**
Resolves #82 

This PR uses the [globals](https://ghub.io/globals) npm module and Electron's [structured API data](https://electronjs.org/blog/api-docs-json-schema) to assemble a list of glossary terms to give translators more context, and help them avoid translating things that should remain untouched.

## What's in the glossary?

- [x] JavaScript builtins like `Array`, `Map`, `String`, etc
- [x] Electron APIs like `BrowserWindow` and `MenuItem`
- [x] Electron API Structures like `MimeTypedBuffer` and `MemoryInfo`
- [x] Electron instance methods like `contents.openDevTools`
- [x] Electron instance properties like `contents.devToolsWebContents`

There's other stuff in both `globals` (like `globals.browser`) and `electron-api.json` (like instance events) that we can squeeze into the glossary, but this is a start.

For those with admin access to Crowdin you can see the glossary entries here: https://crowdin.com/project/electron/settings#glossary

## What translators see
Here's what the glossary entries look like in the translation interface:

screen shot 2018-02-22 at 3 10 18 pm

## Caveats

- The glossary entries are NOT case sensitive, so `buffer` is emphasized in the GUI even though the entry is `Buffer`.
- I ended up choosing wording that accounts for cases where the given word _should_ actually be translated. Hence the wording in the descriptions is "and should usually not be translated".

## Bonus

Whenever this script is run, it updates any existing entries in the glossary, rather than creating duplicate entries. That means we can keep refining and publishing this without worrying about the effect it will have on existing entries.

### CODE REVIEW

**Comment by zeke** [comment]
Made a few updates to separate the building the glossary from uploading. Also added some tests to make sure our glossary data is valid.

**Comment by zeke** [comment]
Stretch goal: Parse https://github.com/electron/electron/blob/master/docs/glossary.md too

File: test/glossary.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    const types = chain(glossary).map('type').uniq().sort().value()
+    types.should.deep.equal([
+      'electronAPI',
+      'electronInstanceMethod',
+      'electronInstanceProperty',
+      'globals.builtin'
+    ])
+  })
+
+  it('does not allow commas in descriptions (for CSV purposes)', () => {
```
Comment (vanessayuenn) [inline test/glossary.js]: 👍

**Comment by zeke** [comment]
Now includes terms from [electron/docs/glossary.md](https://github.com/electron/electron/blob/master/docs/glossary.md) too, like ASAR, DMG, IPC, OSR, Brightray, etc.

screen shot 2018-02-23 at 6 39 02 pm

**Comment by zeke** [comment]
Hey @ckerr and @vanessayuenn I dusted this off and got the tests passing. Can you take another look?

**Pull request merged**